### ISSUE TITLE
Static graphviz task visualisation

### ISSUE THREAD

**Issue by Casyfill** [spotify/luigi#2328]
I am using a small code I wrote for static visualization of Task DAGS using `graphviz` lib. It is currently quite basic, but still I find it quite useful for documentation purposes - I even made a habit of rerendering those plots into task README.md on commit.
I will be happy to share my code via PR, but was wondering if core contributors 

1) think it is a good idea overall
2) have any understanding where to put it, e.g. through `luigi.tools.staticviz` or `luigi.contrib.staticviz`...

**Comment by dlstadther** [comment]
I'm of the option either 
* `luigi.tools` if we want to keep this tightly coupled, or
* as a separate, independent github project

Could you provide some insight into how it is executed against a luigi pipeline?

**Comment by Casyfill** [comment]
Hey @dlstadther, sounds good. I've add a [premature] PR at #2328 with `staticviz` as part of `luigi.tools`.

And the approach is quite simple - for now, I provide it with an endpoint task (last task in the graph), and it scrolls up using `required` method of each consequent node (task). Of course, this approach is somewhat limited, as often we have multiple possible outcomes (thus, multiple endpoint tasks) for one pipeline. I hope to mitigate that in the future, probably just by  creating a separate graph for each endpoint, and then merging the graphs (we can use different edge colors/styles for different pipelines, though).

but before, I have to figure out what went wrong with the tox and cover whole thing with tests

**Comment by EngineerReversed** [comment]
Yes, a static graph visualization would really help. We are working on HPC box and don't have luxury to host servers to visualize graph. A static dump of the dependency graph using `graphviz` would surely help.

**Comment by Casyfill** [comment]
Let me pull and brush my code then. It wasn’t perfect, but worked for me
